src/zope/configuration/config.py

```diff
@@ -148,7 +146,8 @@ def resolve(self, dottedname):
             oname = ''
 
         try:
-            mod = __import__(mname, *_import_chickens)
+            __import__(mname)
+            mod = sys.modules[mname]
```

Comment:
I never liked the _import_chickens magic.  I can't say I like the replacement too much either.

I wonder, could we use `mod = importlib.import_module(mname)` here?

Reply:
`importlib.import_module` is I think possible (and in Python 2 it's more or less identical to the code above with some extra elaboration), and it would solve the problem at hand too.

The difficulty is the exception handling: because the actual import is now one or more levels down the stack, depending on the Python version, in order to distinguish between an `ImportError` caused by the immediate module not existing and something deeper in the stack we'd need to walk through any `ImportError` traceback manually and stop when the frames stop referring to `importlib`.  If people would prefer that (a happy conjunction of making more idiomatic use of modern Python APIs and solving the particular edge case I ran into), then I can certainly have another go, but I'd assumed that it would be too complex a change for the task at hand.

Reply:
Ah, I did not notice the depth checking bits.  `__import__` is fine then.

(I'd add a comment explaining why `mod = __import__(mname)` would not work and why the `sys.modules[]` lookup is necessary, but that's just me.)